src/Orleans/Telemetry/ITelemetryClient.cs

```diff
@@ -0,0 +1,66 @@
+
+        /// <summary>Send a metric value to the registered telemetry consumers.</summary>
+        /// <param name="name">Metric name.</param>
+        /// <param name="value">Metric value.</param>
+        /// <param name="properties">Named string values you can use to classify and filter metrics.</param>
+        void TrackMetric(string name, TimeSpan value, IDictionary<string, string> properties = null);
+
+        /// <summary>Send a metric value of 1 to the registered telemetry consumers, to be displayed in an aggregated view.</summary>
+        /// <param name="name">Metric name.</param>
+        void IncrementMetric(string name);
+
+        /// <summary>Send a metric value of <paramref name="value"/> to the registered telemetry consumers, to be displayed in an aggregated view.</summary>
+        /// <param name="name">Metric name.</param>
+        /// <param name="value">The amount to increment.</param>
+        void IncrementMetric(string name, double value);
+
+        /// <summary>Send a metric value of -1 to the registered telemetry consumers, to be displayed in an aggregated view.</summary>
+        /// <param name="name">Metric name.</param>
+        void DecrementMetric(string name);
+
+        /// <summary>Send a metric value of -<paramref name="value"/> to the registered telemetry consumers, to be displayed in an aggregated view.</summary>
+        /// <param name="name">Metric name.</param>
+        /// <param name="value">The amount to decrement. The value should typically be positive.</param>
+        void DecrementMetric(string name, double value);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackDependency(string name, string commandName, DateTimeOffset startTime, TimeSpan duration, bool success);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackEvent(string name, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackRequest(string name, DateTimeOffset startTime, TimeSpan duration, string responseCode, bool success);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackException(Exception exception, IDictionary<string, string> properties = null, IDictionary<string, double> metrics = null);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackTrace(string message);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackTrace(string message, Severity severityLevel);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackTrace(string message, Severity severityLevel, IDictionary<string, string> properties);
+
+        [Obsolete(TelemetryManager.ObsoleteMessageTelemetry)]
+        void TrackTrace(string message, IDictionary<string, string> properties);
```

Comment:
just one more thing. Isn't we should mark IEventTelemetryConsuemer, ITraceTelemetryConsumer, IExceptionTelemetryConsumer etc, as obsolete too?

Reply:
Since we are deprecating its producing methods. And I think we were in a concensus that we are going to deprecating infrastructure for those telemetry events as a whole.

Reply:
This part I'm not sure, but welcome feedback. I didn't deprecate all methods because if you were actually using other methods, it's because you wanted specifically the APM methods. But if you were using it for tracing just because it was previously conflating the 2 scenarios, I wanted to have a build time warning by obsoleting this. Although by that token I could also osbsolete `TrackException`. Do you have any more thoughts on this? or @sergeybykov ?
With regards to the consumer interfaces themselves, maybe it's a good idea to deprecate `ITraceTelemetryConsumer` at the very least, so if you have a custom implementation of it you know that you might not be getting trace events by default in the future.

Reply:
It's important to note that both metrics and exceptions are correlated. You need exception info/count correlated with some metrics. That way you can have more context and track a whole request with it.

Removing it and leave only traces for exception, will completely loose the context on a given request. Remember, traces are very limited in terms of information/context.